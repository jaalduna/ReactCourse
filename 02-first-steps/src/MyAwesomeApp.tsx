import type { CSSProperties } from "react";

const firstName = "Joaquin";
const lastName = "Aldunate";

const lista = ["a", "b", "c"];
const isActive = true;

const address = { city: "Santiago", zip: "ABC-123" };

const myStyles: CSSProperties = {
  backgroundColor: "lightblue",
  borderRadius: 10,
  padding: 20,
  marginTop: 30,
};

export function MyAwesomeApp() {
  return (
    <>
      <h1 data-testid="first-name-title">{firstName}</h1>
      <h3>{lastName}</h3>
      <p> {lista.join(", ")}</p>
      <h1>{isActive ? "activo" : "no activo"}</h1>
      <p style={myStyles}>{JSON.stringify(address)}</p>
    </>
  );
}
