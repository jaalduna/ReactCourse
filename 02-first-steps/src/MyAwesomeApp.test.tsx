import { describe, expect, test } from "vitest";
import { MyAwesomeApp } from "./MyAwesomeApp";
import { render, screen } from "@testing-library/react";

describe("MyAwesomeApp", () => {
  test("should render first and last name", () => {
    const { container } = render(<MyAwesomeApp />);
    screen.debug();

    const h1 = container.querySelector("h1");
    expect(h1?.innerHTML).toContain("Joaquin");
    const h3 = container.querySelector("h3");
    expect(h3?.innerHTML).toContain("Aldunate");
  });

  test("should render first and last name - screen", () => {
    render(<MyAwesomeApp />);

    const h1 = screen.getByTestId("first-name-title");
  });

  test("should match snapshot", () => {
    const { container } = render(<MyAwesomeApp />);
    expect(container).toMatchSnapshot();
  });
});
