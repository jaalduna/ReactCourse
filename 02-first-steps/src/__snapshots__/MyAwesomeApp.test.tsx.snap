// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <h1
    data-testid="first-name-title"
  >
    Joaquin
  </h1>
  <h3>
    Aldunate
  </h3>
  <p>
     
    a, b, c
  </p>
  <h1>
    activo
  </h1>
  <p
    style="background-color: lightblue; border-radius: 10px; padding: 20px; margin-top: 30px;"
  >
    {"city":"Santiago","zip":"ABC-123"}
  </p>
</div>
`;
